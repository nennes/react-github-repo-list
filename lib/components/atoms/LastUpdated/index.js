/* @flow */

import React from "react";
import { FormattedRelative } from "react-intl";
import { Repo as RepoType } from "../../../types";

const LastUpdated = ({ timestamp }: RepoType) => {
  return (
    <div className="repo__last-updated">
      Last Updated: <FormattedRelative value={timestamp} />
    </div>
  );
};

export default LastUpdated;
