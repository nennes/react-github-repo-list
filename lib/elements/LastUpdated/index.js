import React from "react";
import PropTypes from "prop-types";
import TimeAgo from "react-timeago";

const LastUpdated = ({ timestamp }) => {
  return (
    <div className="repo__last-updated">
      Last Updated: <TimeAgo date={timestamp} />
    </div>
  );
};

LastUpdated.propTypes = {
  timestamp: PropTypes.string.isRequired
};

export default LastUpdated;
