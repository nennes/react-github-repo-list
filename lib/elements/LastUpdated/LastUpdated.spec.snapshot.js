/* global test, expect */

import React from "react";
import LastUpdated from "./index";
import renderer from "react-test-renderer";

test("LastUpdated element renders correctly", () => {
  const element = renderer.create(
    <LastUpdated timestamp="2015-12-13T13:36:22Z" />
  );
  expect(element.toJSON()).toMatchSnapshot();
});
