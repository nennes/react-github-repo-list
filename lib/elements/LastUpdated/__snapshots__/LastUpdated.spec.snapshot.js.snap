// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastUpdated element renders correctly 1`] = `
<div
  className="repo__last-updated"
>
  Last Updated: 
  <time
    dateTime="2015-12-13T13:36:22.000Z"
    title="2015-12-13T13:36:22Z"
  >
    2 years ago
  </time>
</div>
`;
